addon/record-data.js

```diff
@@ -323,7 +327,9 @@ export default class M3RecordData {
       this.id = '' + jsonApiResource.id;
     }
     if (CUSTOM_MODEL_CLASS) {
-      this.globalM3CacheRD[this.id] = this;
+      if (!this._baseRecordData && !this._parentRecordData && this.id) {
```

Comment:
coercion

```suggestion
      if (this._baseRecordData !== undefined && this._parentRecordData !== undefined && this.id !== undefined) {
```

Reply:
commented above lets combine the discussion there